src/sycltest/Makefile

```diff
@@ -0,0 +1,117 @@
+TARGET_DIR := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+TARGET_NAME := $(notdir $(TARGET_DIR))
+TARGET := $(BASE_DIR)/$(TARGET_NAME)
+include Makefile.deps
+EXTERNAL_DEPENDS := $(sycltest_EXTERNAL_DEPENDS)
+
+$(TARGET):
+test_cpu:
+test_sycl: $(TARGET)
```

Comment:
I think I had the command line part in `pixel-standalone`, so eventually I can import it here.
For the moment I am using environment variables to test the various backends, e.g.
```
# use only the SYCL host "dvice"
SYCL_DEVICE_TYPE=HOST ./sycltest

# offload to the CPU using the OpenCL backend
SYCL_DEVICE_TYPE=CPU ./sycltest

# offload to the GPU using the OpenCL backend
SYCL_BE=PI_OPENCL SYCL_DEVICE_TYPE=GPU ./sycltest

# offload to the GPU using the Level 0 backend
SYCL_BE=PI_LEVEL0 SYCL_DEVICE_TYPE=GPU ./sycltest
```

Is there a way to add something like this to the tests, for the time being ?

Reply:
The host "device" could be easily added by
```Makefile
test_cpu:
        @echo
        @echo "Testing $(TARGET)"
        SYCL_DEVICE_TYPE=CPU $(TARGET) --maxEvents 2
        @echo "Succeeded"
```
after which `make test_sycl_cpu` succeeds (assuming https://github.com/cms-patatrack/pixeltrack-standalone/pull/102#discussion_r498563646 is changed as well).

Going beyond that gets tricky. `test_cuda` target makes no sense without CUDA device support, so probably for time being the `test_sycl` is the best we can do. I try to improve (a bit later) how that gets hooked in the top-level `Makefile` (might require a bit more work, because this PR changes the "big picture" quite a lot actually from the earlier "CPU" or "CUDA").

Reply:
Mhm... long term looks like we could have various options:
  - `cpu`: for SYCL, means two options: `host` and `cpu`; also for Alpaka it can map to at least two backends: `serial` and `tbb`
  - `cuda`: for SYCL needs the codeplay and/or open source compiler, I'll look into it in a while
  - `intel` (?):  intel gpu, which can go through two back-ends, OpenCL and Level 0... do we want to test both?
  - `auto` / `all`: all supported backends